pyvo/mivot/features/sky_coord_builder.py

```diff
@@ -0,0 +1,162 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+Utility transforming MIVOT annotation into SKYCoord instances
+"""
+
+from astropy.coordinates import SkyCoord
+from astropy import units as u
+from astropy.coordinates import ICRS, Galactic, FK4, FK5
+
+
+class MangoRoles:
+    """
+    Place holder for the roles of the mango:EpochPosition roles
+    """
+    LONGITUDE = "longitude"
+    LATITUDE = "latitude"
+    PM_LONGITUDE = "pmLongitude"
+    PM_LATITUDE = "pmLatitude"
+    PARALLAX = "parallax"
+    RADIAL_VELOCITY = "radialVelocity"
+    EPOCH = "epoch"
+    FRAME = "frame"
+    EQUINOX = "equinox"
+    PMCOSDELTAPPLIED = "pmCosDeltApplied"
+
+
+# Mapping of the MANGO parameters on the SkyCoord parameters
+skycoord_param_default = {
+    MangoRoles.LONGITUDE: 'ra', MangoRoles.LATITUDE: 'dec', MangoRoles.PARALLAX: 'distance',
+    MangoRoles.PM_LONGITUDE: 'pm_ra_cosdec', MangoRoles.PM_LATITUDE: 'pm_dec',
+    MangoRoles.RADIAL_VELOCITY: 'radial_velocity', MangoRoles.EPOCH: 'obstime'}
+
+skycoord_param_galactic = {
+    MangoRoles.LONGITUDE: 'l', MangoRoles.LATITUDE: 'b', MangoRoles.PARALLAX: 'distance',
+    MangoRoles.PM_LONGITUDE: 'pm_l_cosb', MangoRoles.PM_LATITUDE: 'pm_b',
+    MangoRoles.RADIAL_VELOCITY: 'radial_velocity', MangoRoles.EPOCH: 'obstime'}
+
+
+class SkyCoordBuilder(object):
+    '''
+    Utility generating SkyCoord instances from MIVOT annotations
+
+    - SkyCoord instances can only be built from model classes containing the minimal
+    set of required parameters (a position).
+    - In this implementation, only the mango:EpochPosition class is supported since
+      it proposes anything required to compute the epoch propagation which is a major use-case
+    '''
+
+    def __init__(self, mivot_instance_dict):
+        '''
+        Constructor
+
+        parameters
+        -----------
+        - mivot_instance_dict: viewer.MivotInstance.to_dict()
+                               Internal dictionary of the dynamic Python object generated
+                               from the MIVOT block
+        '''
+        self._mivot_instance_dict = mivot_instance_dict
+        self._map_coord_names = None
+
+    def build_sky_coord(self):
+        """
+        Build a SkyCoord instance from the dynamic Python object attribute.
+
+        returns
+        -------
+        - SkyCoord instance or None
+        """
+        if not self._mivot_instance_dict:
+            return None
+        if self._mivot_instance_dict["dmtype"] == "mango:EpochPosition":
+            return self._build_sky_coord_from_mango()
+        return None
```

Comment:
I don't think you need to return a None here.

Reply:
Some linters (e.g. sonarqube) reuquire method to always return something or never, no mixing is allowed. This is  not the case here, I removed the `return None`